stdlib/asm/crypto/fri.masm

```diff
@@ -0,0 +1,607 @@
+
+    # for j = 61
+	push.8796093020160
+    exec.accumulate_for_odd_index
+
+    # for j = 62
+	push.18446744069414322177
+    exec.accumulate_for_even_index
+
+    # for j = 63
+	push.18302628881338728449
+    exec.accumulate_for_odd_index
+
+    # compute (τ^64 - 1)/ 64
+    dup.3
+    dup.3
+
+    repeat.6
+        dup.1
+        dup.1
+        ext2mul
+    end
+
+    swap
+    sub.1
+    swap
+
+    push.18158513693329981441.0
+    ext2mul
+    ext2mul
+end
+
+#! Given τ and end memory address of interpolated polynomial p ( with degree at max 7, over 
+#! quadratic extension field of Fq ) on stack top, this routine computes α, following formula described
+#! in https://github.com/0xPolygonMiden/miden-vm/issues/592#issue-1481983459
+#!
+#! Expected stack state
+#!
+#! [_, _, τ1, τ0, mem_ptr_p, ...]
+#!
+#! Final stack state
+#!
+#! [α1, α0, _, _, ...]
+#!
+#! Note, mem_ptr_p is absolute memory address of polynomial p ( which was interpolated using advice provider )
+#! s.t. next 3 decreasing ( by 1 ) memory addresses hold remaining 6 coefficients of p. Why this descending order of
+#! memory addressing ? - Read point 2 of https://github.com/0xPolygonMiden/miden-vm/issues/592#issue-1481983459
+proc.compute_alpha
+    movup.4
+    dup
+    sub.1
+    movdn.5
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    dup
+    sub.1
+    movdn.7
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    dup
+    sub.1
+    movdn.7
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    dup.7
+    dup.7
+    ext2mul
+    ext2add
+
+    movup.6
+    push.0.0.0.0
+    movup.4
+    mem_loadw
+
+    movup.5
+    movup.5
+    dup.9
+    dup.9
+    ext2mul
+    ext2add
+
+    movup.7
+    movup.7
+    ext2mul
+    ext2add
+end
+
+#! Given remainder codeword with 64 evaluations ( as starting memory address on stack ), this routine attempts to
+#! probabilistically verify remainder following the technique described in https://github.com/0xPolygonMiden/miden-vm/issues/592
+#! and an optimization of how to efficiently compute β in https://github.com/0xPolygonMiden/miden-vm/pull/644#pullrequestreview-1265141094
+#!
+#! Expected stack state
+#!
+#! [mem_ptr, ...]
+#!
+#! A few assumptions about mem_ptr
+#!
+#! - mem_ptr should be absolute memory address of the beginning of remainder codeword
+#! - Each evaluation is 2 field elements wide because they belong to quadratic extension field over F_p | p = 2^64 - 2^32 + 1
+#! - Meaning each memory address will hold two consecutive evaluations
+#! - Words ( four field elements ), in memory, are laid out in this order (a1_0, a1_0, a0_1, a0_0)
+#! - That means (a0_1, a0_0) -> first evaluation and (a1_1, a1_0) -> next evaluation
```

Comment:
I'm not sure this is correct. The words are stored in memory as `(a0_0, a0_1, a1_0, a1_1)`. It is just that `mem_loadw` loads words onto the stack in little-endian order.

Reply:
Oh correct, looks like I mistyped and didn't catch it. Thanks.